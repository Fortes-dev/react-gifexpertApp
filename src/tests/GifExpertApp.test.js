import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';



describe('Suite GifExpertApp', () => {
    
    test('Debe mostrar el snapshot', () => {
        
        const wrapper = shallow( <GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar una lista de categorias', () => {
        
        const categories = ['One punch', 'Berserk'];

        const wrapper = shallow( <GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });

})