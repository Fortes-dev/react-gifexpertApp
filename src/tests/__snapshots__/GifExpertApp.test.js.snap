// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suite GifExpertApp Debe mostrar el snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Suite GifExpertApp Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Berserk"
      key="Berserk"
    />
  </ol>
</Fragment>
`;
